internal/handler/integration/user_test.go

```diff
@@ -0,0 +1,175 @@
+//go:build integration
+// +build integration
+
+package handler_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/dilyara4949/flight-booking-api/internal/config"
+	"github.com/dilyara4949/flight-booking-api/internal/database/postgres"
+	"github.com/dilyara4949/flight-booking-api/internal/handler"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/request"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/response"
+	"github.com/dilyara4949/flight-booking-api/internal/middleware"
+	"github.com/dilyara4949/flight-booking-api/internal/repository"
+	"github.com/dilyara4949/flight-booking-api/internal/service"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func initDB(cfg config.Config) (*gorm.DB, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	database, err := postgres.Connect(ctx, cfg.Postgres)
+	if err != nil {
+		return nil, fmt.Errorf("database connection failed: %v", err)
+	}
+	return database, nil
+}
+
+func TestUpdateUserHandler(t *testing.T) {
+	cfg, err := config.NewConfig()
+	if err != nil {
+		t.Fatalf("error getting config: %v", err)
+	}
+
+	database, err := initDB(cfg)
+	if err != nil {
+		t.Fatalf("init database failed: %v", err)
+	}
+
+	userRepo := repository.NewUserRepository(database)
+	userService := service.NewUserService(userRepo)
+
+	tests := map[string]struct {
+		userID          string
+		role            string
+		updateReq       request.UpdateUser
+		expectedStatus  int
+		expectedMessage string
+		expectedUser    *response.User
+	}{
+		"OK user role": {
+			userID: "5a57c98d-87a0-436b-a016-634622efbf4e",
+			role:   "user",
+			updateReq: request.UpdateUser{
+				Email: "newuser@mail.ru",
+				Phone: "88888888888",
+			},
+			expectedStatus:  http.StatusOK,
+			expectedMessage: "",
+			expectedUser: &response.User{
+				ID:    uuid.MustParse("5a57c98d-87a0-436b-a016-634622efbf4e"),
+				Email: "newuser@mail.ru",
+				Phone: "88888888888",
+			},
+		},
+		"invalid id": {
+			userID: "5a57c98d-87a0-436b-634622efbf4e",
+			role:   "user",
+			updateReq: request.UpdateUser{
+				Email: "invalid-email",
```

Comment:
this field value is a bit confusing. Since we checking for invalid ID, we don't need invalid email. It would be better to have email validation as a separate test